File: MAIL/mail_server.c
// mailServer.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <unistd.h>
#include <netinet/in.h>

#define PORT 9100
#define BUFFER_SIZE 1024

char stored_mail[BUFFER_SIZE];

void handle_connection(int client_socket) {
    char buffer[BUFFER_SIZE] = {0};
    recv(client_socket, buffer, sizeof(buffer), 0);

    if (strncmp(buffer, "GET", 3) == 0) {
        // IMAP 요청
        send(client_socket, stored_mail, strlen(stored_mail), 0);
    } else {
        // SMTP 요청
        strcpy(stored_mail, buffer); // 메일 저장
        printf("[MailServer] Mail stored: %s\n", stored_mail);
    }

    close(client_socket);
}

int main() {
    int server_fd = socket(AF_INET, SOCK_STREAM, 0);
    struct sockaddr_in addr;
    addr.sin_family = AF_INET;
    addr.sin_port = htons(PORT);
    addr.sin_addr.s_addr = INADDR_ANY;

    bind(server_fd, (struct sockaddr*)&addr, sizeof(addr));
    listen(server_fd, 5);
    printf("[MailServer] Running on port %d...\n", PORT);

    while (1) {
        int client_socket = accept(server_fd, NULL, NULL);
        handle_connection(client_socket);
    }

    return 0;
}

File: MAIL/mail_server.py
# mail_server.py
import socket
import threading

def handle_mail_client(conn, addr):
    print(f"[MailServer] Connected with: {addr}")
    while True:
        data = conn.recv(1024)
        if not data:
            break
        print(f"[MailServer] Received SMTP message: {data.decode()}")
        conn.sendall(b"250 OK")  # SMTP 성공 응답
    conn.close()

def run_mail_server():
    HOST = '127.0.0.1'
    PORT = 2525 # SMTP 포트

    with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as server:
        server.bind((HOST, PORT))
        server.listen()
        print("[MailServer] Waiting for SMTP connection...")
        while True:
            conn, addr = server.accept()
            threading.Thread(target=handle_mail_client, args=(conn, addr)).start()

if __name__ == '__main__':
    run_mail_server()
